src/introspection/introspection.ts

```diff
@@ -202,6 +202,46 @@ function markDeprecated(schema: SimplifiedIntrospectionWithIds): void {
   // which are deprecated.
 }
 
+function markInterfaceFields(schema: SimplifiedIntrospectionWithIds): void {
+  function isFieldOnInterfaceSameAsOnType(objectField, interfaceField) {
```

Comment:
@IvanGoncharov added check that fields are same. I tested it manually with this schema
```graphql
    interface IFoo {
      name: String
    }
    type Foo implements IFoo {
      name: String
    }
    type Bar {
      name: String
    }
    union FooAndBar = Foo | Bar

    # same types
    interface I0 {
      name: Foo
    }
    type Example0_hide implements I0 {
      name: Foo
    }

    # possible type of interface
    interface I1 {
      name: IFoo
    }
    type Example11_hide implements I1 {
      name: IFoo
    }
    type Example12 implements I1 {
      name: Foo
    }

    # possible type of union
    interface I2 {
      name: FooAndBar
    }
    type Example21_hide implements I2 {
      name: FooAndBar
    }
    type Example22 implements I2 {
      name: Foo
    }
    type Example23 implements I2 {
      name: Bar
    }

    # list subtyping
    interface I3 {
      name: [IFoo]
    }
    type Example31_hide implements I3 {
      name: [IFoo]
    }
    type Example32 implements I3 {
      name: [Foo]
    }

    # !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
    # stricker nullability
    # !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!

    # same types
    interface I4 {
      name: Foo
    }
    type Example4 implements I4 {
      name: Foo!
    }

    # possible type of interface
    interface I5 {
      name: IFoo
    }
    type Example51 implements I5 {
      name: IFoo!
    }
    type Example52 implements I5 {
      name: Foo!
    }

    # same union
    interface I6 {
      name: FooAndBar
    }
    type Example6 implements I6 {
      name: FooAndBar!
    }

    # possible type of union
    interface I7 {
      name: FooAndBar
    }
    type Example71 implements I7 {
      name: Foo!
    }
    type Example72 implements I7 {
      name: Bar!
    }

    # list subtyping
    interface I8 {
      name: [IFoo]
    }
    type Example81 implements I8 {
      name: [IFoo!]
    }
    type Example82 implements I8 {
      name: [Foo!]
    }
    type Example83 implements I8 {
      name: [IFoo]!
    }
    type Example84 implements I8 {
      name: [Foo]!
    }
    type Example85 implements I8 {
      name: [IFoo!]!
    }
    type Example86 implements I8 {
      name: [Foo!]!
    }

    # list subtyping 2
    interface I9 {
      name: [IFoo!]
    }
    type Example91_hide implements I9 {
      name: [IFoo!]
    }
    type Example92 implements I9 {
      name: [Foo!]
    }
    type Example93 implements I9 {
      name: [IFoo!]!
    }
    type Example94 implements I9 {
      name: [Foo!]!
    }

    # list subtyping 3
    interface I10 {
      name: [IFoo]!
    }
    type Example101_hide implements I10 {
      name: [IFoo]!
    }
    type Example102 implements I10 {
      name: [Foo]!
    }
    type Example103 implements I10 {
      name: [IFoo!]!
    }
    type Example104 implements I10 {
      name: [Foo!]!
    }

    # argument list
    interface I11 {
      name(some: String): Foo
    }
    type Example111_hide implements I11 {
      name(some: String): Foo
    }
    type Example112 implements I11 {
      name(some: String, aditionalArg:String): Foo
    }
    

    

    union Result =
        Example0_hide
      | Example11_hide
      | Example12
      | Example21_hide
      | Example22
      | Example23
      | Example31_hide
      | Example32
      | Example4
      | Example51
      | Example52
      | Example6
      | Example71
      | Example72
      | Example81
      | Example82
      | Example83
      | Example84
      | Example85
      | Example86
      | Example91_hide
      | Example92
      | Example93
      | Example94
      | Example101_hide
      | Example102
      | Example103
      | Example104
      | Example111_hide
      | Example112
      
    type Query {
      getFoo: Result
      getFooBar: FooAndBar
      # Get one todo item
    }

    type Mutation {
      addTodo(name: String!): String!
    }

    schema {
      query: Query
      mutation: Mutation
    }
```